pandas/tests/test_index.py

```diff
@@ -2755,10 +2789,27 @@ def test_get_loc(self):
             self.assertEqual(idx.get_loc(idx[1], method), 1)
             self.assertEqual(idx.get_loc(idx[1].to_pydatetime(), method), 1)
             self.assertEqual(idx.get_loc(str(idx[1]), method), 1)
+            self.assertEqual(idx.get_loc(idx[1], method,
+                                         tolerance=pd.Timedelta('0 days')), 1)
 
         self.assertEqual(idx.get_loc('2000-01-01', method='nearest'), 0)
         self.assertEqual(idx.get_loc('2000-01-01T12', method='nearest'), 1)
 
+        self.assertEqual(idx.get_loc('2000-01-01T12', method='nearest',
+                                     tolerance='1 day'), 1)
+        self.assertEqual(idx.get_loc('2000-01-01T12', method='nearest',
+                                     tolerance=pd.Timedelta('1D')), 1)
+        self.assertEqual(idx.get_loc('2000-01-01T12', method='nearest',
+                                     tolerance=np.timedelta64(1, 'D')), 1)
+        self.assertEqual(idx.get_loc('2000-01-01T12', method='nearest',
+                                     tolerance=timedelta(1)), 1)
+        # TODO: fix this (it currently raises KeyError)
+        # with tm.assertRaises(ValueError):
+        #     idx.get_loc('2000-01-01T12', method='nearest', tolerance='foo')
+        with tm.assertRaises(KeyError):
+            idx.get_loc('2000-01-01T03', method='nearest',
```

Comment:
nope, this is a legit `KeyError` (2 hours is not enough tolerance for 3 hours offset)